catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/QueryResultIterable.java

```diff
@@ -65,8 +68,7 @@ public QueryResultIterable(CatalogFramework catalog, Function<Integer, QueryRequ
      *                 the caller can control iteration based on their own logic
      * @param pageSize How many results should each page hold
      */
-    public QueryResultIterable(CatalogFramework catalog, Function<Integer, QueryRequest> filter,
-            int pageSize) {
+    public QueryResultIterable(CatalogFramework catalog, Filter filter, int pageSize) {
```

Comment:
I like the idea of being able to specify a guaranteed result-size (it would be good to not over-query for too many results if they only want very something that is 1 result over a new query boundry).
Though, Why do you think it would be better split into two separate classes instead of just an overloaded ctor for this that either allows them to specify max and then the other can hide the page-size/result-size all together.